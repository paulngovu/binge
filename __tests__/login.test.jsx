/*
 * Testing for login page
 */

import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Login from '../pages/login';
import {
  TESTID_LOGIN_BUTTON,
  TESTID_LOGIN_PASSWORD,
  TESTID_LOGIN_TITLE,
  TESTID_LOGIN_USERNAME,
} from '../testIds';

describe('Login page', () => {
  let tree;
  let getByTestId;

  beforeEach(() => {
    tree = renderer.create(<Login />).toJSON();
    ({ getByTestId } = render(<Login />));
  });

  test('Renders entire page', () => {
    expect(tree).toMatchSnapshot();
  });

  test('Renders title', () => {
    expect(getByTestId(TESTID_LOGIN_TITLE)).toBeTruthy();
  });

  test('Renders username field', () => {
    expect(getByTestId(TESTID_LOGIN_USERNAME)).toBeTruthy();
  });

  test('Renders password field', () => {
    expect(getByTestId(TESTID_LOGIN_PASSWORD)).toBeTruthy();
  });

  test('Renders login button', () => {
    expect(getByTestId(TESTID_LOGIN_BUTTON)).toBeTruthy();
  });
});
