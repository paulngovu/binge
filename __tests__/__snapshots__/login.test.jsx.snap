// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Renders entire page 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 hjInHh"
>
  <header
    className="StyledBox-sc-13pk1d4-0 gpGtYH"
    display="flex"
  >
    <h1
      className="StyledHeading-sc-1rdh4aw-0 elclaf"
    >
      Binge
    </h1>
    <div
      className="StyledBox__StyledBoxGap-sc-13pk1d4-1 itCgil"
    />
    <div
      alignself="end"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 wrGSV"
      />
    </div>
  </header>
  <div
    className="StyledBox-sc-13pk1d4-0 hHCfST"
  >
    <div
      className="jsx-1619348227 container"
    >
      <span
        className="StyledText-sc-1sadyjn-0 flpFwE"
        data-testid="login-title"
      >
        Login
      </span>
      <form
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="StyledBox-sc-13pk1d4-0 cImcDU FormField__FormFieldBox-sc-m9hood-0 bNDIlh"
          data-testid="login-username"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="StyledText-sc-1sadyjn-0 gvmMDy"
          >
            username
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 ctACHi FormField__FormFieldContentBox-sc-m9hood-1"
          >
            <div
              className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 jqwtzG"
            >
              <input
                autoComplete="off"
                className="StyledTextInput-sc-1x30a0s-0 bmcLpb"
                name="username"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="username"
              />
            </div>
          </div>
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 cImcDU FormField__FormFieldBox-sc-m9hood-0 bNDIlh"
          data-testid="login-password"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <label
            className="StyledText-sc-1sadyjn-0 gvmMDy"
          >
            password
          </label>
          <div
            className="StyledBox-sc-13pk1d4-0 ctACHi FormField__FormFieldContentBox-sc-m9hood-1"
          >
            <div
              className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 jqwtzG"
            >
              <input
                autoComplete="off"
                className="StyledTextInput-sc-1x30a0s-0 bmcLpb"
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                placeholder="password"
                type="password"
              />
            </div>
          </div>
        </div>
        <span
          className="StyledText-sc-1sadyjn-0 bGUCcJ"
        >
          Don't have an account? Sign up 
          <a
            className="jsx-1619348227"
            href="/register"
          >
            here
          </a>
          .
        </span>
        <div
          className="StyledBox-sc-13pk1d4-0 gXhMXE"
        >
          <button
            className="StyledButton-sc-323bzc-0 emnQnE"
            data-testid="login-btn"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="submit"
          >
            Log In
          </button>
        </div>
        <span
          className="StyledText-sc-1sadyjn-0 fBFkhO"
          data-testid="login-error-msg"
        >
          
        </span>
      </form>
    </div>
  </div>
</div>
`;
